packages/block-library/src/button/index.php

```diff
@@ -17,7 +17,15 @@
  * @return string The block content.
  */
 function render_block_core_button( $attributes, $content ) {
-	$p = new WP_HTML_Tag_Processor( $content );
+	/*
+	 * The current Gutenberg plugin supports WordPress 6.4, but the next_token()
+	 * method does not exist in WordPress 6.4. Therefore, use the Gutenberg
+	 * class that has the `next_token()` method.
+	 *
+	 * TODO: After the Gutenberg plugin drops support for WordPress 6.4, this
+	 * class should be replaced with `WP_HTML_Tag_Processor`.
+	 */
+	$p = new Gutenberg_HTML_Tag_Processor_6_5( $content ); // phpcs:ignore Gutenberg.CodeAnalysis.ForbiddenFunctionsAndClasses.ForbiddenClassUsage
```

Comment:
what's the ForbiddenClassUsage warning here?
is this something new?


I want to make sure we don't sneak through the package import process accidentally.

@ellatrix do you have insight here?

Reply:
This sniff seems to have been introduced in #57738.

Without this comment, PHPCS will emit the following warning.

```
It's not allowed to use the "Gutenberg_HTML_Tag_Processor_6_5" class as its name matches the forbidden pattern: "/^[Gg]utenberg.*$/".
```

On the other hand, I found that no warning is displayed if the code is like this:

```php
if ( defined( 'IS_GUTENBERG_PLUGIN' ) && IS_GUTENBERG_PLUGIN && class_exists( 'Gutenberg_HTML_Tag_Processor_6_5' ) ) {
	$p = new Gutenberg_HTML_Tag_Processor_6_5( $content );
} else {
	$p = new WP_HTML_Tag_Processor( $content );
}
```

Reply:
@anton-vlasenko can you comment on this?

Reply:
👋 @dmsnell,
PHP code in the `block-library` folder must not use `(G|g)utenberg` prefixed classes or functions, as these blocks are backported to the WordPress release package, where Gutenberg classes/functions are not available.
This rule doesn't apply to experimental blocks since they are not backported.
Gutenberg prefixed classes/methods should only be used if the Gutenberg plugin is available, ensured by the `if ( defined( 'IS_GUTENBERG_PLUGIN' ) && IS_GUTENBERG_PLUGIN )` check.
Related discussion: https://github.com/WordPress/gutenberg/pull/54509#discussion_r1332282682

Reply:
thanks @anton-vlasenko - this is the first time I've seen code ported into Core that checks if it's running the plugin and then changes its behavior. I guess this is intentional?

Reply:
Yes, this is intentional (not sure if I got your question right) in the sense that this is needed to avoid fatal errors in the core. There are several such checks in WordPress Core: https://github.com/search?q=repo%3AWordPress%2Fwordpress-develop++%22if+%28+defined%28+%27IS_GUTENBERG_PLUGIN%27+%29+%26%26+IS_GUTENBERG_PLUGIN%22&type=code"

Reply:
very good. I thought we were rewriting these calls when importing into Core instead of bleeding the plugin code into it. thanks for confirming that this was intentional.